### PR TITLE
create RPC tests for sell and buy calculations

### PR DESCRIPTION
**Pull request by goncer**
Create buy/sell calculations that checks:

- Zero and negative values and validation handling
- Two accuracy tests
- Test that ensures that the amount calculated is the real amount resturned after buy/sell

### CODE REVIEW

File: test/parallel/xyk-rpc.error.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	).rejects
+	.toThrow(expected.toString())
+	
+});
+
+test.each([
+	[new BN(1),new BN(1),new BN(0),new BN(0)],
+	[new BN(0),new BN(0),new BN(0),new BN(0)],
+	[new BN(0),new BN(1),new BN(1),new BN(1)],
+	[new BN(0),new BN(0),new BN(1),new BN(0)],
```
Comment (iStrike7) [inline test/parallel/xyk-rpc.error.test.ts:43]: The pool can not exist in these states. So they are not semantically significant, but I guess you can use them to test if the rpc is behaving as expected (or some subtle behaviour has changed).

File: test/parallel/xyk-rpc.calculate.test.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	
+	const numberOfAssets = new BN(100);
+	let sellPriceRpc = await calculate_sell_price_rpc(poolBalanceBefore[0], poolBalanceBefore[1], numberOfAssets);
+	await testUser1.sellAssets(firstCurrency, secondCurrency , numberOfAssets);
+	await testUser1.refreshAmounts(AssetWallet.AFTER);
+	const assetsSold = testUser1.getAsset(firstCurrency)?.amountAfter;
+	const assetsBought = testUser1.getAsset(secondCurrency)?.amountAfter;
+
+	//the relation of buy and sell is maintained.
+	//because of rounding, we need to expend one unit more
```
[the remark this answers is not in the document]
Comment (goncer) [reply test/parallel/xyk-rpc.calculate.test.ts:113]: you are right! removing!

File: test/parallel/xyk-rpc.calculate.test.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	
+	const numberOfAssets = new BN(100);
+	let sellPriceRpc = await calculate_buy_price_rpc(poolBalanceBefore[0], poolBalanceBefore[1], numberOfAssets);
+	await testUser1.buyAssets(firstCurrency, secondCurrency , numberOfAssets);
+	await testUser1.refreshAmounts(AssetWallet.AFTER);
+	const assetsSold = testUser1.getAsset(firstCurrency)?.amountAfter;
+	const assetsBought = testUser1.getAsset(secondCurrency)?.amountAfter;
+
+	//the relation of buy and sell is maintained.
+	//because of rounding, we need to expend one unit more
```
[the remark this answers is not in the document]
Comment (goncer) [reply test/parallel/xyk-rpc.calculate.test.ts:131]: removing!

File: test/parallel/xyk-rpc.error.test.ts (hunk 1 above)
Comment (goncer) [reply test/parallel/xyk-rpc.error.test.ts:43]: that's the idea!

**Review by iStrike7 — APPROVED** [review]

**Pull request merged**